### PR TITLE
feat: support shadow dom selectors

### PR DESCRIPTION
**Pull request by michael-siek**
Closes: https://github.com/dequelabs/axe-core-npm/issues/615

### CODE REVIEW

**Review by stephenmathieson — CHANGES_REQUESTED** [review]
Why was https://github.com/dequelabs/axe-core-npm/actions/workflows/update-axe-core.yml not used?

**Comment by michael-siek** [comment]
Because we also needed to add shadow dom selector changes

**Comment by stephenmathieson** [comment]
> Because we also needed to add shadow dom selector changes

That should be in its own commit.

File: packages/playwright/src/index.ts [hunk 1]
```diff
@@ -148,6 +151,7 @@ export default class AxeBuilder {
   public async analyze(): Promise<AxeResults> {
     const context = normalizeContext(this.includes, this.excludes);
     const { page, option: options } = this;
+    console.log(JSON.stringify(context, null, 2));
```
Comment (straker) [inline packages/playwright/src/index.ts:154]:
```suggestion
```

File: packages/webdriverjs/src/utils/index.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 /**
  * Get running context
  */
 export const normalizeContext = (
-  include: Selector[],
-  exclude: Selector[]
+  include: SerialSelectorList,
+  exclude: SerialSelectorList
 ): ContextObject => {
   const base: ContextObject = {
```
Comment (straker) [inline packages/webdriverjs/src/utils/index.ts:10]: Hu, does this work? I thought we had to make everything a `SerialContextObject`

**Review by straker — DISMISSED** [review]
The type changes and tests all look correct.

File: packages/playwright/src/index.ts [hunk 3]
```diff
@@ -16,11 +20,12 @@ import {
   axeShadowSelect
 } from './browser';
 import AxePartialRunner from './AxePartialRunner';
+type NewContextObject = ContextObject | SerialContextObject;
```
Comment (straker) [inline packages/playwright/src/index.ts:23]: This doesn't appear to be used anywhere

```suggestion
```

**Review by straker — CHANGES_REQUESTED** [review]

**Review by straker — APPROVED** [review]

**Pull request merged**